core/api.go

```diff
@@ -141,30 +89,52 @@ type EntryView interface {
 	TTL() time.Duration
 }
 
+// AbstractMapEvent is base for a map event.
+type AbstractMapEvent interface {
```

Comment:
I couldnt find the usage for `AbstractMapEvent` on the client side. Can you show where it is used on the client side?

Reply:
In Java server side, some interfaces are declared and used in client side. Otherwise, same interface will be created twice. Therefore, that there is no `AbstractMapEvent` on the client side is normal.

Reply:
Here `AbstractMapEvent` is an interface. On Java side, it is named abstract because it is an abstract class. 
I think it is okay to have `MapEvent` and `EntryEvent` derive from a base interface since they have common methods but naming is not right. I would change its name to `Event`.